### PR TITLE
classes: mfgtool-initramfs-image use weak variable for IMAGE_FSTYPES

### LINKED ISSUE #951: fitImage: fsl-image-mfgtool-initramfs doesn't allow ramfs
Hi,

I am trying to generate a fitImage from `fsl-image-mfgtool-initramfs`. After a `loadaddr` issue already reported in meta-freescale issue, I am struggling with the initramfs.

As the `its` file wasn't reporting the ramdisk, I tried to put some configurations.

### INITRAMFS Configurations

From **local.conf**

```
KERNEL_CLASSES  = " kernel-fitimage"
KERNEL_IMAGETYPES = "fitImage zImage"
IMAGE_FSTYPES += "tar.bz2 cpio.gz cpio.gz.u-boot"
INITRAMFS_IMAGE = "fsl-image-mfgtool-initramfs"
INITRAMFS_IMAGE_BUNDLE = "1"
```
meta:  **honister**
meta-freescale: **honister**

### Behaviour

```
ERROR: Task (/workdir/meta-custom/recipes-kernel/linux/linux-fslc-lts-mfgtool_5.10.bb:do_bundle_initramfs) failed with exit code '1'
ERROR: linux-fslc-lts-5.10.91+gitAUTOINC+aa8525ec05-r0 do_bundle_initramfs: Could not find any /workdir/build/tmp/deploy/images/imx6ull-custom/fsl-image-mfgtool-initramfs-imx6ull-custom.cpio{.gz|.lz4|.lzo|.lzma|.xz|.zst) for bundling; INITRAMFS_IMAGE_NAME might be wrong.
```

The current behaviour is that the file **cpio.gz** is not present in the deploy directory and can't be added to the fitImage.

Did I miss something?

### Current related work

From https://patchwork.openembedded.org/patch/67525/

I made one step foward by removing the overwrite in `mfgtool-initramfs-image.bbclass`. 

But, now, I have two rootfs bundled (kernel+ramfs, ramfs). I have a corner case or bad intent with the current state of the art.

```
IMAGE_FSTYPES = "cpio.gz.u-boot"
```
<details close>
  <summary>Log boot fitImage-initramfs</summary>

```
## Loading kernel from FIT Image at 82000000 ...
   Using 'conf-imx6ull-dart6ul-6ulcustomboard.dtb' configuration
   Trying 'kernel-1' kernel subimage
     Description:  Linux kernel
     Type:         Kernel Image
     Compression:  uncompressed
     Data Start:   0x82000128
     Data Size:    19983416 Bytes = 19.1 MiB
     Architecture: ARM
     OS:           Linux
     Load Address: 0x80008000
     Entry Point:  0x80008000
     Hash algo:    sha256
     Hash value:   1bb3d953ee38e80d386a843195b1e8dca04f6a56c47fa276f9f12f96ead426b9
   Verifying Hash Integrity ... sha256+ OK
## Loading ramdisk from FIT Image at 82000000 ...
   Using 'conf-imx6ull-dart6ul-6ulcustomboard.dtb' configuration
   Trying 'ramdisk-1' ramdisk subimage
     Description:  fsl-image-mfgtool-initramfs
     Type:         RAMDisk Image
     Compression:  uncompressed
     Data Start:   0x83316278
     Data Size:    8773298 Bytes = 8.4 MiB
     Architecture: ARM
     OS:           Linux
     Load Address: unavailable
     Entry Point:  unavailable
     Hash algo:    sha256
     Hash value:   345f8d1b2c588bf50648b058d8c682dba0a358b207fdc150e4806be238ac964e
   Verifying Hash Integrity ... sha256+ OK
## Loading fdt from FIT Image at 82000000 ...
   Using 'conf-imx6ull-dart6ul-6ulcustomboard.dtb' configuration
   Trying 'fdt-imx6ull-dart6ul-6ulcustomboard.dtb' fdt subimage
     Description:  Flattened Device Tree blob
     Type:         Flat Device Tree
     Compression:  uncompressed
     Data Start:   0x8330ee80
     Data Size:    29477 Bytes = 28.8 KiB
     Architecture: ARM
     Hash algo:    sha256
     Hash value:   707a722a1ba092d00a22bb17fa419ae810fcb29c3a7090bb1d269c5b7c6a278e
   Verifying Hash Integrity ... sha256+ OK
   Booting using the fdt blob at 0x8330ee80
   Loading Kernel Image
Image too large: increase CONFIG_SYS_BOOTM_LEN
Must RESET board to recover
resetting ...
```
</details>

### Maybe related issue

I have seen a very unlucky bug like that before : 
- https://www.yoctoproject.org/pipermail/yocto/2017-August/037454.html
- https://bugzilla.yoctoproject.org/show_bug.cgi?id=11928

### PR DESCRIPTION
**Pull request by IniterWorker**
Close #951 by changing `IMAGE_FSTYPES` to a weak variable

### CODE REVIEW

**Review by otavio — CHANGES_REQUESTED** [review]
Please extend the commit log explaining the reasoning for that.

**Comment by IniterWorker** [comment]
Follow up, I don't know if the force push notifies you @otavio. Do you need additional information?

**Review by otavio — APPROVED** [review]

**Comment by otavio** [comment]
@IniterWorker can you prepare the backport for the branches (using cherry-pick -x) as it failed automated backport?

File: classes/mfgtool-initramfs-image.bbclass [hunk 1]
```diff
@@ -13,8 +13,8 @@ FEATURE_PACKAGES_extfs = "packagegroup-fsl-mfgtool-extfs"
 FEATURE_PACKAGES_f2fs = "packagegroup-fsl-mfgtool-f2fs"
 
 ZSTD_COMPRESSION_LEVEL ?= "-10"
-IMAGE_FSTYPES = "cpio.zst.u-boot"
-IMAGE_FSTYPES:mxs-generic-bsp = "cpio.gz.u-boot"
+IMAGE_FSTYPES ?= "cpio.zst.u-boot"
```
Comment (thochstein) [inline classes/mfgtool-initramfs-image.bbclass:16]: Unfortunately this change makes this line a no-op and breaks the default case.

```
# $IMAGE_FSTYPES [6 operations]
#   set? /.../sources/meta-freescale/conf/machine/include/imx-base.inc:463
#     "${SOC_DEFAULT_IMAGE_FSTYPES}"
#   set /.../sources/poky/meta/conf/documentation.conf:214
#     [doc] "Formats of root filesystem images that you want to have created."
#   set? /.../sources/poky/meta/conf/bitbake.conf:832
#     "tar.gz"
#   set? /.../sources/meta-freescale/classes/mfgtool-initramfs-image.bbclass:16
#     "cpio.zst.u-boot"
#   override[mxs-generic-bsp]:set? /.../sources/meta-freescale/classes/mfgtool-initramfs-image.bbclass:17
#     "cpio.gz.u-boot"
# pre-expansion value:
#     "${SOC_DEFAULT_IMAGE_FSTYPES}"
IMAGE_FSTYPES="wic.bmap  wic.bz2 tar.bz2"
```

Comment (otavio) [reply classes/mfgtool-initramfs-image.bbclass:16]: We can override the SOC_DEFAULT_IMAGE_FSTYPES; it would work.

Comment (thochstein) [reply classes/mfgtool-initramfs-image.bbclass:16]: Okay, see https://github.com/Freescale/meta-freescale/pull/1110.

**Pull request merged**